lib/Controller/AccountsController.php

```diff
@@ -421,14 +420,13 @@ public function send(int $id,
 		}
 
 		$messageData = NewMessageData::fromRequest($account, $expandedTo, $expandedCc, $expandedBcc, $subject, $body, $attachments, $isHtml, $requestMdn);
-		$repliedMessageData = null;
 		if ($messageId !== null) {
 			try {
 				$repliedMessage = $this->mailManager->getMessage($this->currentUserId, $messageId);
+				$repliedMessageId = $repliedMessage->getMessageId();
 			} catch (ClientException $e) {
 				$this->logger->info("Message in reply " . $messageId . " could not be loaded: " . $e->getMessage());
 			}
-			$repliedMessageData = new RepliedMessageData($account, $repliedMessage);
```

Comment:
what the old data structure and identification allowed was to receive a message with account A, reply with account B and still get the flag updates on account A. This will break now.

Reply:
Raised as https://github.com/nextcloud/mail/issues/6189